components/applications-service/pkg/storage/postgres/service_groups.go

```diff
@@ -270,3 +275,25 @@ func queryFromStatusFilter(text string) (string, error) {
 		return "", errors.Errorf("invalid status filter '%s'", text)
 	}
 }
+
+// formatSortFields returns a customized ORDER BY statement from the provided sort field,
+// if the field doesn't require a special ORDER BY it will return the same field
+//
+// Criteria:
+// * When sort by percent_ok, add a second order by group health criticality.
+// * When sort by group health criticality, add a second order by percent_ok.
+//
+// The second ORDER BY added with this function will be static and it will always be ascending
+// since we want to always give more priority to the things that are in a critical state
+func formatSortFields(field, sort string) string {
+	sortField := field + " " + sort
+
+	switch field {
+	case "health":
```

Comment:
I don't think we sort by health, isn't it always percent ok and then health within like percentages?

Reply:
I talked with @apriofrost about enabling both ways of sorting since now we can do deep sorting in those fields, we agreed in the acceptance criteria that are detailed in this PR as well as in the Jira Ticket. *This is NOT the default behavior though*, it is just an optional item we wanted to cover. This is not harming or compromising anything but instead, enable us to do other kinds of sorting. Let us know if this is a blocker from your side, we can talk about it in standup. 👍

Reply:
Yeah, the default sorting is by percentage OK. We enable the second option in the case that our current assumption is proved not correct, and users would prefer health more than the percentage. So that we can adjust the UI design (e.g. flip the position of the mini donut chart and the health status icon) and switch the default sorting order to the second option.